lib/Doctrine/ORM/EntityManager.php

```diff
@@ -860,7 +860,7 @@ protected static function createConnection($connection, Configuration $config, E
         }
 
         if ( ! $connection instanceof Connection) {
-            throw new \InvalidArgumentException("Invalid argument: " . $connection);
+            throw new \InvalidArgumentException(sprintf('Invalid argument for connection "%s".', is_object($connection) ? get_class($connection) : gettype($connection) . '#' . $connection));
```

Comment:
The `#` is confusing in my opinion. The message should probably be `'Invalid $connection argument of type %s given: "%s"'`